ssl/statem/statem_lib.c

```diff
@@ -1688,18 +1688,21 @@ int tls_get_message_body(SSL_CONNECTION *s, size_t *len)
          * The TLsv1.3 handshake transcript stops at the ClientFinished
          * message.
          */
-#define SERVER_HELLO_RANDOM_OFFSET  (SSL3_HM_HEADER_LENGTH + 2)
+        const size_t hdr_len = SSL_CONNECTION_IS_DTLS(s) ? DTLS1_HM_HEADER_LENGTH
+                                                         : SSL3_HM_HEADER_LENGTH;
+        const size_t srvhellorandom_offs = hdr_len + 2;
+
         /* KeyUpdate and NewSessionTicket do not need to be added */
         if (!SSL_CONNECTION_IS_VERSION13(s)
             || (s->s3.tmp.message_type != SSL3_MT_NEWSESSION_TICKET
                          && s->s3.tmp.message_type != SSL3_MT_KEY_UPDATE)) {
             if (s->s3.tmp.message_type != SSL3_MT_SERVER_HELLO
-                    || s->init_num < SERVER_HELLO_RANDOM_OFFSET + SSL3_RANDOM_SIZE
+                    || s->init_num < srvhellorandom_offs + SSL3_RANDOM_SIZE
                     || memcmp(hrrrandom,
-                              s->init_buf->data + SERVER_HELLO_RANDOM_OFFSET,
+                              s->init_buf->data + srvhellorandom_offs,
                               SSL3_RANDOM_SIZE) != 0) {
                 if (!ssl3_finish_mac(s, (unsigned char *)s->init_buf->data,
-                                     s->init_num + SSL3_HM_HEADER_LENGTH)) {
+                                     s->init_num + hdr_len, 1)) {
```

Comment:
Hmm made this agnostic to whatever protocol runs it, but it looks like it is only run for TLS. And so I was looking at the comparable code for DTLS and I couldn't find this check. I guess that is a mistake. My next thought is to have a common function that is called instead of duplicating code. Would that be correct or am I missing something?

Reply:
Yes I agree. I was only focused on this minor bit. I'll do that then.